locksmith/src/graphql/datasource/keysByQuery.ts

```diff
@@ -153,27 +153,53 @@ export class keysByQuery extends GraphQLDataSource {
   }: KeyGetProps) {
     try {
       const first = 1000 // max items
-      const skip = parseInt(`${page}`, 10) * first
+
       const expireTimestamp = parseInt(`${new Date().getTime() / 1000}`)
       const keyId = getValidNumber(search)
 
-      let query
+      let query: any
       if (filterKey === 'keyId' && `${search}`?.length) {
         query = QUERY_BY_TYPE.keyId
       } else {
         query = QUERY_BY_TYPE[expiration]
       }
 
-      const response = await this.query(query, {
-        variables: {
-          addresses,
-          keyId,
-          first,
-          skip,
-          expireTimestamp,
-        },
-      })
-      return response.data.locks
+      const getData = async (getFromPage = page) => {
+        const skip = parseInt(`${getFromPage}`, 10) * first
+        return await this.query(query, {
+          variables: {
+            addresses,
+            keyId,
+            first,
+            skip,
+            expireTimestamp,
+          },
+        })
+      }
+      const {
+        data: { locks },
+      } = (await getData()) ?? {}
+
+      const keysList = locks[0]?.keys ?? []
+
+      let getForNextPage = keysList?.length === first
```

Comment:
it's unfortunate that total isn't working on current subgraph otherwise we could use that for detecting whether there are more keys. Maybe add a comment here for future that we have no way to detect how many keys there are left to query on this subgraph query.

There should be cursor on the graphql side.